### PR TITLE
chore: add release for v3 templates

### PR DESCRIPTION
**Pull request by yukun-dong**

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -194,6 +210,18 @@ jobs:
...
           removeArtifacts: true
 
+      - name: release v3 templates' RC version to github
+        if: ${{ (contains(steps.version-change.outputs.CHANGED, 'templates@') || contains(steps.version-change.outputs.CHANGED, '@microsoft/teamsfx')) && github.event_name == 'workflow_dispatch' && github.event.inputs.preid == 'rc' }}
+        uses: ncipollo/release-action@v1.10.0
+        with:
+          token: ${{ secrets.CD_PAT }}
+          prerelease: true
+          name: "templatesv3@0.0.0-rc"
+          tag: "templates@0.0.0-rc"
```
Comment (hund030) [inline .github/workflows/cd.yml:220]: Seems it conflicts to current rc template?

Comment (yukun-dong) [reply .github/workflows/cd.yml:220]: Indeed, I will use a new tag here.

File: .github/scripts/template-zip-autogen-v3.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+for TEMPLATE in ${TEMPLATE_LIST[@]}; do
+
+    if [ ! -d ./templates/scenarios/${TEMPLATE} ]; then
+        echo "The folder ./templates/scenarios/${TEMPLATE}  does not exist."
+        exit -1
+    fi
+
+    cd ./templates/scenarios/${TEMPLATE} 
+    zip -rq ${TEMPLATE}.zip .
```
Comment (hund030) [inline .github/scripts/template-zip-autogen-v3.sh:26]: `zip -rq  ${TEMPLATE_OUTPUT_DIR}/${TEMPLATE}.zip .`

Comment (yukun-dong) [reply .github/scripts/template-zip-autogen-v3.sh]: Thanks! Fixed.

**Review by hund030 — DISMISSED** [review]
LGTM, better to test it in your fork repo.

File: .github/scripts/template-zip-autogen-v3.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
```
Comment (wenytang-ms) [inline .github/scripts/template-zip-autogen-v3.sh:1]: why do not merge to template-zip-autogen.sh?

**Comment by yukun-dong** [comment]
> LGTM, better to test it in your fork repo.

https://github.com/yukun-dong/TeamsFx/actions/runs/3326693483

**Review by hund030 — APPROVED** [review]

File: packages/fx-core/src/component/generator/utils.ts [hunk 4]
```diff
@@ -36,16 +36,17 @@ const sampleTagPrefix = sampleConfig.tagPrefix;
...
-  }
-  if (preRelease === "beta") {
-    return templateBetaVersion;
-  }
-  const versionPattern = preRelease ? `0.0.0-${preRelease}` : templateVersion;
-  const versionList = tags.map((tag: string) => tag.replace(templateTagPrefix, ""));
-  const selectedVersion = semver.maxSatisfying(versionList, versionPattern);
-  return selectedVersion ? templateTagPrefix + selectedVersion : undefined;
+  return templateAlphaVersion;
+  // if (preRelease === "alpha") {
```
Comment (wenytang-ms) [inline packages/fx-core/src/component/generator/utils.ts:40]: why not delete directly?

Comment (yukun-dong) [reply packages/fx-core/src/component/generator/utils.ts:40]: These code are commented out temporarily. When v3 is ready they will be uncommented.

Comment (yukun-dong) [reply packages/fx-core/src/component/generator/utils.ts:40]: This method is only called by v3. For v2 it still uses old ways to match tags.

File: templates/package.json [hunk 5]
```diff
@@ -45,6 +45,24 @@
         "tab/js/m365",
         "tab/ts/m365"
     ],
+    "templatesV3": [
```
Comment (wenytang-ms) [inline templates/package.json:48]: seem the sync-up script need to update. 
these tabs also need to sync the sdk version
you can ask @hund030 for more info

Comment (hund030) [reply templates/package.json:48]: Good points! I think it's ok to update the sync-up script later and ensure the release process works.

Comment (yukun-dong) [reply templates/package.json:48]: Create a follow up task here: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/16017883

**Review by wenytang-ms — APPROVED** [review]

**Pull request merged**